plugins/out_cef/cef_config.c

```diff
@@ -0,0 +1,772 @@
+{"request",                     "requestUrl",                  CEF_STR, 1023},
+{"requestClientApplication",    "requestClientApplication",    CEF_STR, 1023},
+{"requestContext",              "requestContext",              CEF_STR, 2048},
+{"requestCookies",              "requestCookies",              CEF_STR, 1023},
+{"requestMethod",               "requestMethod",               CEF_STR, 1023},
+{"rt",                          "deviceReceiptTime",           CEF_TIME,0   },
+{"shost",                       "sourceHostName",              CEF_STR, 1023},
+{"smac",                        "sourceMacAddress",            CEF_MAC, 0   },
+{"sntdom",                      "sourceNtDomain",              CEF_STR, 255 },
+{"sourceDnsDomain",             "sourceDnsDomain",             CEF_STR, 255 },
+{"sourceServiceName",           "sourceServiceName",           CEF_STR, 1023},
+{"sourceTranslatedAddress",     "sourceTranslatedAddress",     CEF_IPV4,0   },
+{"sourceTranslatedPort",        "sourceTranslatedPort",        CEF_INT, 0   },
+{"spid",                        "sourceProcessId",             CEF_INT, 0   },
+{"spriv",                       "sourceUserPrivileges",        CEF_STR, 1023},
+{"sproc",                       "sourceProcessName",           CEF_STR, 1023},
+{"spt",                         "sourcePort",                  CEF_INT, 0   },
+{"src",                         "sourceAddress",               CEF_IPV4,0   },
+{"start",                       "startTime",                   CEF_TIME,0   },
+{"suid",                        "sourceUserId",                CEF_STR, 1023},
+{"suser",                       "sourceUserName",              CEF_STR, 1023},
+{"type",                        "type",                        CEF_INT, 0   },
+{NULL,                          NULL,                          0,       0   }
+};
+
+static struct cef_cdic cef_cdic_ipv6 =
+    { "c6a%dLabel",              1023, "c6a%d",                0, CEF_IPV6 };
+static struct cef_cdic cef_cdic_fp =
+    { "cfp%dLabel",              1023, "cfp%d",                0, CEF_FLOAT};
+static struct cef_cdic cef_cdic_number =
+    { "cn%dLabel",               1023, "cn%d",                 0, CEF_LONG };
+static struct cef_cdic cef_cdic_string =
+    { "cs%dLabel",               1023, "cs%d",              4000, CEF_STR  };
+static struct cef_cdic cef_cdic_date =
+    { "deviceCustomDate%dLabel", 1023, "deviceCustomDate%d",   0, CEF_TIME };
+static struct cef_cdic cef_cdic_flex_date =
+    { "flexDate%dLabel",          128, "flexDate%d",           0, CEF_TIME };
+static struct cef_cdic cef_cdic_flex_str =
+    { "flexString%dLabel",        128, "flexString%d",      1023, CEF_STR  };
+
+
+static int cef_ht_add (struct cef_ht *ht, flb_sds_t key,
+                       struct cef_ht_dic *dic, int dic_size,
+                       struct cef_ht *child);
+
+/*
+ * This hash generation function is taken originally from Redis source code:
```

Comment:
why our hashing functions are not suitable?

Reply:
I was assuming it was only for strings, because of the mallocs of the key and the value and the use that it was currently being given. Checking it again I was wrong. It Complicate things a bit by the malloc when creating an entry but it is not a problem to change it to the fluentbit implementation.